### ISSUE TITLE
Difference in color palette (256 colors) between FreeRDP (Windows) and native rdp client

### ISSUE THREAD

**Issue by ToM-MaS** [FreeRDP/FreeRDP-old#27]
Hi everyone, 
I'm currently developing the new Remote Desktop Client for ReactOS ( http://www.reactos.org ).
That means I'm using the FreeRDP Windows version.
I'm connecting from Windows 7 to a Windows 2000 Server for this example. (means 8 bit, 256 colors)
I've noticed a color difference between FreeRDP (git version 2011-05-02) and the native mstsc.
This image shows the difference (left freerdp, right mstsc - look at the mspaint color palette at the bottom):
http://www.thomasbrandt.net/ReactOS/FreeRDP/Freerdp_palette_problem.png

That's all.
Keep up the good work.
(Please don't neglect xrdp too much, ReactOS will need it some day :) )

**Comment by awakecoding** [comment]
Hi,

The ReactOS project is welcome to replace rdesktop by FreeRDP for their built-in RDP client ;)

I'm willing to give it a try and see if I can find the source of the problem, but I do not have a Windows 2000 Server machine at my disposal. Were you able to reproduce the same issue with a more recent version of Windows?

**Comment by ToM-MaS** [comment]
Hi!
Yeah, it's the same when I connect to Windows XP SP3 with 8 bit color depth preset in both clients.
http://www.thomasbrandt.net/ReactOS/FreeRDP/Freerdp_palette_problem_XP.png

**Comment by otavio** [comment]
Can't this issue has been fixed at with the change done in #50?

**Comment by awakecoding** [comment]
@otavio: no, it has to do with palette handling, gdi_mono_image_convert() shouldn't be related to this

**Comment by dorianj** [comment]
I can confirm this on Unix using libfreerdp-gdi, I see it with a WinXP SP3 host in 8bit mode.

**Comment by otavio** [comment]
If you can come up with a fix for it, it would be nice :-)

**Comment by dorianj** [comment]
I'm going to work on it. I have worked with the color palettes in rdesktop before.
